### ISSUE TITLE
[Bug]: publish copies DLL config files

### ISSUE THREAD

**Issue by MisinformedDNA** [dotnet/msbuild#9011]
### Issue Description

`dotnet publish` copies config files from dependent libraries instead of just the config files from the published app.

### Steps to Reproduce

1. Create a class library (ClassLibrary1) and add an `app.config` file (Build Action: "None", Copy to Output Directory: "Do not copy").
2. Create a console app and reference the class library.
3. Run `dotnet publish -c Release` or `msbuild -t:publish -p:Configuration=Release` on the console project

### Expected Behavior

Only config files for the published app are copied.

### Actual Behavior

ClassLibrary1.dll.config is published.

### Analysis

As I understand it, those config files, by default, are ignored. So, why are they there?

### Versions & Configurations

❯ msbuild --version
MSBuild version 17.6.3+07e294721 for .NET Framework
17.6.3.22601

❯ dotnet --version
8.0.100-preview.4.23260.5

[PublishConfig.zip](https://github.com/dotnet/msbuild/files/12017592/PublishConfig.zip)

**Comment by AR-May** [comment]
Team triage: This is by design. If one has some file associated with dll, it is a reasonable default is to bring it along during publish. If some configuration file was created by a developer, it is probably because it is going to be used, so we need to bring it. Also, `app.config` files were important for .NET framework, so there are plenty of libraries that would not work without it.

**Comment by MisinformedDNA** [comment]
> even though WCF will try to read them from the exe config file only, and ignore the dll config file

This is true for all .NET Framework code I've migrated to .NET (Core); all the extra config files are completely ignored. So, while publishing the config files might make sense for .NET Framework, I still don't see why they are published for .NET.
